platform/pc/service.c

```diff
@@ -187,16 +188,49 @@ static void new_cpu()
         kernel_sleep();
 }
 
+closure_function(0, 2, void, count_processors_handler,
+                 u8, type, void *, p)
+{
+    switch (type) {
+    case ACPI_MADT_LAPIC:
+        if (((acpi_lapic)p)->flags & MADT_LAPIC_ENABLED)
+            present_processors++;
+        break;
+    case ACPI_MADT_LAPICx2:
+        if (((acpi_lapic_x2)p)->flags & MADT_LAPIC_ENABLED)
+            present_processors++;
+        break;
+    }
+}
+
+static void count_processors()
+{
+    acpi_madt madt = acpi_get_table(ACPI_SIG_MADT);
+    if (madt) {
+        present_processors = 0;
+        acpi_walk_madt(madt, stack_closure(count_processors_handler));
+        init_debug("ACPI reports %d processors", present_processors);
+    } else {
+        present_processors = 1;
+        init_debug("ACPI MADT not found, default to 1 processor");
```

Comment:
Should the MADT be available in every installation? If so, maybe this message should be printed unconditionally as a warning. This way we can detect if there's an oddball platform that isn't conforming and might be running single core when multiple are specified.